language/language-tests/result.rs

```diff
@@ -56,3 +56,34 @@ pub fn match_result() -> SimpleOutput {
         SResult::Err(_) => SimpleOutput::new(),
     }
 }
+
+fn no_annotation() -> Result<u8, u16> {
+    let _intermediate: Result<u8, u16> = Ok(123u8);
+    Err(1u16)
+}
+
+fn no_annotation_nested() -> Result<u8, u16> {
+    if 1 > 3 {
+        Err(1u16)
+    } else {
+        match Result::<u8, u16>::Ok(12u8) {
+            Result::<u8, u16>::Ok(x) => Err(x as u16),
+            Result::<u8, u16>::Err(_x) => Ok(1u8),
+        }
+    }
+}
+
+fn no_annotation_tuple() -> (Result<(), u16>, Result<u8, u16>) {
+    if 1 > 3 {
+        (Ok(()), Err(1u16))
+    } else {
+        match Result::<u8, u16>::Ok(12u8) {
+            Result::<u8, u16>::Ok(x) => (Ok(()), Err(x as u16)),
+            Result::<u8, u16>::Err(_x) => (Ok(()), Ok(1u8)),
+        }
+    }
+}
+
+fn no_annotation_fun_call() -> U32 {
```

Comment:
Can you add a test with a `?`? Just another case where I can see bugs 🙂

Reply:
Sure, just commited some more tests in that direction.
I also added some type propagation when typechecking constructors.